### PR TITLE
Migrate to poetry

### PR DESCRIPTION
**Pull request by hfjn**
Since pipenv hasn't seen an update for over a [year](https://github.com/real-digital/esque/compare/migrate-to-poetry) we decided to switch to poetry.

### CODE REVIEW

File: esque/__init__.py [hunk 1]
```diff
@@ -0,0 +1,5 @@
+import configparser
+
+parser = configparser.ConfigParser()
+parser.read("pyproject.toml")
```
Comment (swenzel) [inline esque/__init__.py]: Is this actually independent from the working directory? Could be that we have to use path relative to `__file__` here.

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,57 @@
+# This file is part of Poetry
```
Comment (swenzel) [inline Makefile]:
```suggestion
# This file was part of Poetry
```

File: Makefile [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+# https://github.com/sdispater/poetry
+
+# Licensed under the MIT license:
+# http://www.opensource.org/licenses/MIT-license
+# Copyright (c) 2018 Sébastien Eustace
+
+POETRY_RELEASE := $$(sed -n -E "s/__version__ = '(.+)'/\1/p" esque/__init__.py)
```
Comment (swenzel) [inline Makefile]:
```suggestion
```
That variable isn't used anywhere...

File: pyproject.toml [hunk 4]
```diff
@@ -1,4 +1,55 @@
...
+version = "0.1.4a"
+description="esque - an operational kafka tool."
+authors = ["real.digital <opensource@real-digital.de>"]
+license = "MIT"
+readme = "README.md"
+exclude = ["tests/**/*"]
+
+repository = "https://github.com/real.digital/esque"
+
+keywords = ["kafka", "dependency", "poetry"]
```
Comment (swenzel) [inline pyproject.toml]:
```suggestion
keywords = ["kafka", "cli", "commandline", "administration", "operation"]
```

File: pyproject.toml [hunk 5]
```diff
@@ -1,4 +1,55 @@
...
+
+classifiers = [
+  "Development Status :: 3 - Alpha",
+  "Intended Audience :: Developers",
```
Comment (swenzel) [inline pyproject.toml]:
```suggestion
  "Intended Audience :: System Administrators and Developers",
```

File: pyproject.toml [hunk 6]
```diff
@@ -1,4 +1,55 @@
...
+  "Topic :: Software Development :: Build Tools",
```
Comment (swenzel) [inline pyproject.toml]:
```suggestion
  "Topic :: System Administration :: Operational Tools",
```

File: tests/unit/test_ruleparser.py [hunk 7]
```diff
@@ -0,0 +1,103 @@
+from unittest import TestCase
```
Comment (swenzel) [inline tests/unit/test_ruleparser.py]: I don't think this file belongs here.

File: esque/__init__.py (hunk 1 above)
Comment (hfjn) [reply esque/__init__.py]: Hm. Yeah it's probably safer to get `__init__` file's path and start from there.

File: tests/unit/test_ruleparser.py (hunk 7 above)
Comment (hfjn) [reply tests/unit/test_ruleparser.py]: I just moved it here. For some reason it was located in `esque/ruleparser/tests/unit`. Don't know why github doesn't recognize that.

File: Makefile [hunk 8]
```diff
@@ -0,0 +1,56 @@
+# This file was part of Poetry
```
Comment (hfjn) [inline Makefile]: Maybe we get just rid of this header? It's MIT licensed anyway.

Comment (swenzel) [reply Makefile]: Was actually more meant as a joke :D yeah get rid of it... we don't use licence headers anyway.

File: tests/unit/test_ruleparser.py (hunk 7 above)
Comment (swenzel) [reply tests/unit/test_ruleparser.py]: You have to delete `esque/ruleparser/tests`

Comment (hfjn) [reply tests/unit/test_ruleparser.py]: ah goddamnit

**Review by swenzel — APPROVED** [review]

**Pull request merged**